components/public-api/gitpod/experimental/v1/oidc.proto

```diff
@@ -140,8 +145,11 @@ service OIDCService {
   // Updates modifiable properties of an existing OIDC client configuration.
   rpc UpdateClientConfig(UpdateClientConfigRequest) returns (UpdateClientConfigResponse) {};
 
-  // Removes a OIDC client configuration by ID.
+  // Removes an OIDC client configuration by ID.
   rpc DeleteClientConfig(DeleteClientConfigRequest) returns (DeleteClientConfigResponse) {};
+
+  // Activates an OIDC client configuration by ID.
+  rpc ActivateClientConfig(ActivateClientConfigRequest) returns (ActivateClientConfigResponse) {};
```

Comment:
What's the reasoning for making this an explicit method rather than using UpdateClientConfig which sets the value?

We wanted to keep the OIDCService mostly Resource based, but this makes it action based. This also means we'd need to add `DeactiveClientConfig` the moment we want to support the symmetrical action on here, rather than setting the value of `config.active = false`.

Reply:
Discussed this point with @selfcontained, and we thought that the validation rules checked for this action could be difficult to combine and then consume from frontend. 

The principle ideas are:
 * (for now) have a single SSO config at a time
 * you can only activate a verified config
 * verification won't create a new session
 * activation of one config will deactivate the other (if any)
 * modifying an active config should be blocked
 * create a new config, then verify it, and then activate to avoid breaking sign-ins
 * de-activating of the single active one should be blocked
   * de-activating in general would become interesting with multi IdP support (out of scope for now)
 

> We wanted to keep the OIDCService mostly Resource based, but this makes it action based. 

I'm also fine with using `UpdateClientConfig`. The disadvantage in doing so, might be the complexity of dealing with multiple changes in a single update request, e.g. `UpdateClientConfig({ active: true, secret: "xyz" })`. Yes we can internally check if `active` is to be changed.

Reply:
I'd personally prefer sticking with Update, as it ensures we don't grow the number of RPCs on the Public API with the number of attributes we have on the resource. Happy to pair on the change if you'd like

Reply:
The way to fix the "is the value not set, or is it false" we can do the following:

In the .proto definition:
```
message Foo {
  optional bool verified = 1;
}
```
The `optional` modifier effectively makes the generated value be a pointer bool, which now allows you to assert whether it was set, or not. This is a backwards compatible change as well.
```
Verified *bool `protobuf:"varint,5,opt,name=oauth_only,json=oauthOnly,proto3,oneof" json:"oauth_only,omitempty"`
```

Reply:
Don't have a super strong opinion on this, but I also think increasing the # of api methods to handle specific scenarios is a tradeoff worth making, especially in this case. The side-effects, validation and parameters are so different between update and activate that we'd essentially be shoving 2 methods into 1, and making maintenance more difficult in some ways.